### PR TITLE
Highcharts - Add maxFontSize and minFontSize to series that have it.

### PR DESCRIPTION
**Pull request by justinmwarner**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://api.highcharts.com/highcharts/series.line.label.maxFontSize
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by justinmwarner** [comment]
@DanielRosenwasser , sorry, I didn't see that my previous PR had any comments.

Old PR: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29954

In response to:

> Are all of these just part of SeriesCharts? If not, why not extract them to their own interface and avoid the duplication?

I replied:

> According to the docs, this is only available for a subset of these, so I went through the docs and updated the appropriate ones instead of doing all of them.

For the tslint, how do I enable this? The project has a tslint.json file and I did run the command mentioned in the PR template.

File: types/highcharts/index.d.ts [hunk 1]
```diff
@@ -4644,6 +4644,14 @@ declare namespace Highcharts {
          * allows setting a separate color for the line without altering the fillColor.
          */
         lineColor?: string | Gradient;
+        /**
+         * For area-like series, allow the font size to vary so that small areas get a smaller font size. The default applies this effect to area-like series but not line-like series.
+         */
+        maxFontSize?: number;
```
Comment (uniqueiniquity) [inline types/highcharts/index.d.ts]: In the [docs link you posted](https://api.highcharts.com/highcharts/series.line.label.maxFontSize), it states that the default value is `null`. Should the type therefore be `number | null` in each of these cases?

**Review by uniqueiniquity — CHANGES_REQUESTED** [review]

**Comment by uniqueiniquity** [comment]
@justinmwarner Regarding Daniel's suggestion to enable lint rules, you should remove the lines in the `tslint.json` that are disabling specific rules. The most important one is the `dtheader` one, but it would be good to remove all of them.

**Comment by justinmwarner** [comment]
I've done this to ones that are managable. The others have many problems that I think would be best done in a separate PR by someone who may be more familiar with how they work.

---
In reply to: [440363369](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30412#issuecomment-440363369) [](ancestors = 440363369)

**Comment by uniqueiniquity** [comment]
@justinmwarner Looks like you might need to fix the `no-angle-bracket-type-assertion` violations. That should be a straightforward fix since it's just a small syntax change. Thanks!

**Comment by uniqueiniquity** [comment]
@justinmwarner You might need to update the version of `dtslint` that you're running locally. That type assertion rule is one that was added recently, and that's distributed through the dtslint package. I had the same problem myself last week. 

Thanks for all your work getting this into shape!

**Review by uniqueiniquity — APPROVED** [review]

**Pull request merged**